mlflow/store/artifact/utils/models.py

```diff
@@ -32,39 +37,59 @@ def _get_latest_model_version(client, name, stage):
+    version: Optional[str] = None
+    stage: Optional[str] = None
+    alias: Optional[str] = None
+
+
 def _parse_model_uri(uri):
     """
-    Returns (name, version, stage). Since a models:/ URI can only have one of
-    {version, stage, 'latest'}, it will return
-        - (name, version, None) to look for a specific version,
-        - (name, None, stage) to look for the latest version of a stage,
-        - (name, None, None) to look for the latest of all versions.
+    Returns a ParsedModelUri tuple. Since a models:/ URI can only have one of
+    {version, stage, 'latest', alias}, it will return
+        - (name, version, None, None) to look for a specific version,
+        - (name, None, stage, None) to look for the latest version of a stage,
+        - (name, None, None, None) to look for the latest of all versions.
+        - (name, None, None, alias) to look for a registered model alias.
     """
     parsed = urllib.parse.urlparse(uri)
     if parsed.scheme != "models":
         raise MlflowException(_improper_model_uri_msg(uri))
-
     path = parsed.path
-    if not path.startswith("/") or len(path) <= 1:
+    r = re.compile(r"^\/(?P<model_name>[\w \-]+)(?P<suffix>\/[\w]+)?(?P<alias>@[\w\-]+)?$")
+    m = r.match(path)
+    if m is None:
         raise MlflowException(_improper_model_uri_msg(uri))
-    parts = path[1:].split("/")
-
-    if len(parts) != 2 or parts[0].strip() == "":
+    model_name = m.group("model_name") if "model_name" in m.groupdict() else None
+    suffix = m.group("suffix") if "suffix" in m.groupdict() else None
+    alias = m.group("alias") if "alias" in m.groupdict() else None
+    if (
+        (model_name is None or model_name == "")
+        or (suffix and alias)
+        or (suffix is None and alias is None)
+    ):
         raise MlflowException(_improper_model_uri_msg(uri))
 
-    if parts[1].isdigit():
+    if alias:
+        # The URI is an alias URI, e.g. "models:/AdsModel1@Champion"
+        return ParsedModelUri(model_name, alias=alias[1:])
+    suffix = suffix[1:]
+    if suffix.isdigit():
         # The suffix is a specific version, e.g. "models:/AdsModel1/123"
-        return parts[0], int(parts[1]), None
-    elif parts[1].lower() == _MODELS_URI_SUFFIX_LATEST.lower():
+        return ParsedModelUri(model_name, version=suffix)
+    elif suffix.lower() == _MODELS_URI_SUFFIX_LATEST.lower():
         # The suffix is the 'latest' string (case insensitive), e.g. "models:/AdsModel1/latest"
-        return parts[0], None, None
+        return ParsedModelUri(model_name)
     else:
         # The suffix is a specific stage (case insensitive), e.g. "models:/AdsModel1/Production"
-        return parts[0], None, parts[1]
+        return ParsedModelUri(model_name, stage=suffix)
 
 
 def get_model_name_and_version(client, models_uri):
-    (model_name, model_version, model_stage) = _parse_model_uri(models_uri)
+    (model_name, model_version, model_stage, model_alias) = _parse_model_uri(models_uri)
```

Comment:
With the namedtuple implementation, can't we assign the return result to a single variable and then use dot notation to access the required elements?

i.e.
```
if model_version is not None:
  return parsed_uri.name, parsed_uri.version
```

Reply:
Yeah we could but in that example it would be 
```
if parsed_uri.version is not None:
  return parsed_uri.name, parsed_uri.version
```
It just felt more verbose and bloated the code imo

Reply:
I'd choose the current code in this case :). If you just need a single field, `parse_uri.name` makes sense, but if you need all the fields, unpacking them into each variable would look cleaner.